db/views/reporting_facility_states_v08.sql

```diff
@@ -0,0 +1,254 @@
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'dead') AS dead,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state != 'dead') AS cumulative_assigned_patients
+
+        FROM reporting_patient_states
+        WHERE hypertension = 'yes'
+        GROUP BY 1, 2
+    ),
+
+    assigned_diabetes_patients AS (
+        SELECT assigned_facility_region_id AS region_id, month_date,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care') AS diabetes_under_care,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up') AS diabetes_lost_to_follow_up,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'dead') AS diabetes_dead,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state != 'dead') AS cumulative_assigned_diabetes_patients
+
+        FROM reporting_patient_states
+        WHERE diabetes = 'yes'
+        GROUP BY 1, 2
+    ),
+
+    adjusted_outcomes AS (
+        SELECT assigned_facility_region_id AS region_id, month_date,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care' AND htn_treatment_outcome_in_last_3_months = 'controlled') AS controlled_under_care,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care' AND htn_treatment_outcome_in_last_3_months = 'uncontrolled') AS uncontrolled_under_care,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care' AND htn_treatment_outcome_in_last_3_months = 'missed_visit') AS missed_visit_under_care,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care' AND htn_treatment_outcome_in_last_3_months = 'visited_no_bp') AS visited_no_bp_under_care,
+
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up' AND htn_treatment_outcome_in_last_3_months = 'missed_visit') AS missed_visit_lost_to_follow_up,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up' AND htn_treatment_outcome_in_last_3_months = 'visited_no_bp') AS visited_no_bp_lost_to_follow_up,
+
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care') AS patients_under_care,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up') AS patients_lost_to_follow_up
+
+        FROM reporting_patient_states
+        WHERE hypertension = 'yes'
+          AND months_since_registration >= 3
+        GROUP BY 1, 2
+    ),
+
+    adjusted_diabetes_outcomes AS (
+        SELECT assigned_facility_region_id AS region_id, month_date,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care' AND diabetes_treatment_outcome_in_last_3_months = 'bs_below_200' AND blood_sugar_type = 'random') AS random_bs_below_200_under_care,
```

Comment:
I missed this in the previous PR, can you confirm that `blood_sugar_type` will contain the reading type of _the most recent_ BS as of the reporting month, right?

Reply:
@harimohanraj89 Yes, blood_sugar_type is the type of the most recent blood sugar measurement as of a given month. 

The reporting pipeline is similar to that of blood pressures.